auction-app-backend/src/main/java/com/example/auctionapp/controller/ProductController.java

```diff
@@ -0,0 +1,78 @@
+package com.example.auctionapp.controller;
+
+import com.example.auctionapp.model.Product;
+import com.example.auctionapp.service.ProductService;
+import com.example.auctionapp.request.ProductAddRequest;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Sort;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/api/v1/products")
+public class ProductController {
+
+    private final ProductService productService;
+
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @GetMapping
+    public Page<Product> getProductsPaginated(
```

Comment:
Returning paginated data is standard approach, there is no need to name this methods as `getProductsPaginated`
`getProducts` works perfectly fine